include/villas/fpga/card.hpp

```diff
@@ -93,11 +93,11 @@ class PCIeCard : public Card {
 	lookupIp(const ip::IpIdentifier &id) const;
 
 	bool
-	mapMemoryBlock(const MemoryBlock &block);
+	mapMemoryBlock(std::shared_ptr<MemoryBlock> block);
 
 private:
 	// Cache a set of already mapped memory blocks
-	std::set<MemoryManager::AddressSpaceId> memoryBlocksMapped;
+	std::map<MemoryManager::AddressSpaceId, std::shared_ptr<MemoryBlock>> memoryBlocksMapped;
```

Comment:
I dont see why you need a map here. Where is the reference to the blocks of the map?

I only see `mappedMemoryBlock.first` no `mappedMemoryBlock.second`.

Reply:
Also consider using `MemoryBlock::Ptr`